### PR TITLE
[DOC-3619] Migrate existing delegate to mTLS

### PR DESCRIPTION
**Pull request by brian-f-harness**
Migrate existing delegate to mTLS

For reviewers, a previews is available.

[Move delegates to mTLS](https://64ed01cd1261220bda6419ef--harness-developer.netlify.app/docs/platform/Delegates/secure-delegates/delegate-mtls-support#move-delegates-to-mtls)

## What Type of PR is This?

- [ ] Issue
- [x] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [x] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

**Review by pratmit — DISMISSED** [review]
LGTM

File: docs/platform/2_Delegates/secure-delegates/delegate-mtls-support.md [hunk 1]
```diff
@@ -71,11 +71,138 @@ In the following examples, OpenSSL is used to create the required certificates.
...
-Download the new YAML file from the Harness UI and update the certificates in it:
+You can move existing delegates with an immutable image type (image tag `yy.mm.xxxxx`) to mTLS.
 
-```yaml
-client.crt:
-client.key:
-```
+### Prerequisites for delegate migration
+
+Make sure to meet the following prerequisites before you migrate an existing delegate to an mTLS-enabled delegate.
```
Comment (douglas-j-bothwell) [inline docs/platform/2_Delegates/secure-delegates/delegate-mtls-support.md:80]:
```suggestion
Before you migrate an existing delegate to an mTLS-enabled delegate, make sure that you meet the following prerequisites.
```

File: docs/platform/2_Delegates/secure-delegates/delegate-mtls-support.md [hunk 2]
```diff
@@ -71,11 +71,138 @@ In the following examples, OpenSSL is used to create the required certificates.
...
+        mountPath: /etc/mtls
+        readOnly: true
+        value: "/etc/mtls/client.crt"
+   volumes:
+   - name: client-certificate
+     secret:
+       secretName: kubernetes-delegate-client-certificate
+   ```
+
+6. Add the client certificate and key section in the `ConfigMap` `upgrader-config`, and add your `managerHost` and port.
```
Comment (douglas-j-bothwell) [inline docs/platform/2_Delegates/secure-delegates/delegate-mtls-support.md:139]:
```suggestion
6. Add the client certificate and key section in the `ConfigMap` `upgrader-config`, then add your `managerHost` and port. 
```

**Review by douglas-j-bothwell — DISMISSED** [review]
A few word-nerd sugs but otherwise looks good.

**Pull request merged**